pinot-common/src/main/java/com/linkedin/pinot/common/utils/DataSize.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (C) 2014-2015 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.utils;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Utility to convert human readable datsize strings like '4567GB', '128MB'
+ * to bytes
+ */
+public class DataSize {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DataSize.class);
+
+  static final Pattern STORAGE_VAL_PATTERN = Pattern.compile("([\\d.]+)([TGMK]?B)", Pattern.CASE_INSENSITIVE);
```

Comment:
Use a case sensitive match, to avoid confusion between "Mb" (megabit) and "MB" (megabyte).